relay-dynamic-config/src/defaults.rs

```diff
@@ -70,7 +132,7 @@ pub fn add_span_metrics(project_config: &mut ProjectConfig) {
                 key: key.into(),
                 field: Some(format!("span.data.{}", key.replace('.', "\\."))),
                 value: None,
-                condition: None,
+                condition: Some(additional_conditions.clone()),
```

Comment:
You do not need to add conditions to the tag mappings. The metrics themselves are already extracted conditionally, and tag mapping only applies on the extracted mapping.